From 751fed6e0799a553b324addc9b1140d481cddfbb Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Fri, 15 May 2015 14:22:03 +0800
Subject: [PATCH] MLK-10894 pci: imx: Error message and kernel dump happens
 after resume

phenomena:
There is no such kind of phenomen when the "no_consol_suspend"
is added into kernel cmd line.
But there maybe kernel dump when the "no_consol_suspend" is
removed from kernel cmd line.
Root cause:
After the RC resume back and link is up, delay for a while is
required to let ep to resume from D3 mode
Regarding to RM DOC, no status bit can be used to make a
check on it, thus a while delay had been added.
Test results: Over weekend stress tests are passed.

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
(cherry picked from commit 1046f1eb20979cdd874d99601ee889ddf26caf51)
---
 drivers/pci/host/pci-imx6.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/pci/host/pci-imx6.c b/drivers/pci/host/pci-imx6.c
index 564f0e1..f7ede14 100644
--- a/drivers/pci/host/pci-imx6.c
+++ b/drivers/pci/host/pci-imx6.c
@@ -1203,6 +1203,8 @@ static int pci_imx_resume_noirq(struct device *dev)
 		ret = imx6_pcie_wait_for_link(pp);
 		if (ret < 0)
 			pr_info("pcie link is down after resume back.\n");
+		/* Delay for a while, give ep some times to resume from D3 */
+		udelay(200);
 	} else {
 		/*
 		 * L2 can exit by 'reset' or Inband beacon (from remote EP)
-- 
1.8.0

